src/stream/src/executor/join/builder.rs

```diff
@@ -156,6 +157,29 @@ impl<const T: JoinTypePrimitive, const SIDE: SideTypePrimitive> JoinChunkBuilder
         }
     }
 
+    pub fn post_process(c: StreamChunk) -> StreamChunk {
+        let mut c = StreamChunkMut::from(c);
+
+        // NOTE(st1page): remove the pattern `UpdateDel(k, old), UpdateIns(k, NULL), UpdateDel(k, NULL),  UpdateIns(k, new)`
+        // to avoid this issue <https://github.com/risingwavelabs/risingwave/issues/17450>
+        let mut i = 1;
+        while i < c.capacity() {
+            if c.op(i - 1) == Op::UpdateInsert
+                && c.op(i) == Op::UpdateDelete
```

Comment:
LGTM, but I recommend adding an assertion (or check) for row i-2 and i+1 e.g.

```suggestion
            if c.op(i - 2) == Op::UpdateDelete
                && c.op(i - 1) == Op::UpdateInsert
                && c.op(i) == Op::UpdateDelete
                && c.op(i + 1) == Op::UpdateInsert
```